plutus-pab/sync-client/Main.hs

```diff
@@ -33,46 +36,40 @@ slotConfig =
 
 -- | A simple callback that reads the incoming data, from the node.
 processBlock
-  :: BlockInMode CardanoMode
+  :: ChainSyncEvent
   -> Slot
   -> IO ()
-processBlock (BlockInMode (Block (BlockHeader (SlotNo slot) hsh _) _) _) _ =
-  putStrLn $ "Received block " <> show hsh
+processBlock (RollForward (BlockInMode (Block (BlockHeader (SlotNo slot) hsh _) _) _)) _ =
+  putStrLn $ "Received block " <> show (serialiseToRawBytesHexText hsh)
           <> " for slot "      <> show slot
+processBlock (Resume slot hash) _ =
+  putStrLn $ "Resuming from slot " <> show slot
+          <> " at hash " <> show hash
 
-getBlockHash
-  :: CardanoBlock StandardCrypto
-  -> String
-getBlockHash = \case
-  BlockByron blk ->
-    show $ unByronHash $ blockHash blk
-  BlockShelley (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockAllegra (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockMary (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockAlonzo (ShelleyBlock _ headerHash) ->
-    show headerHash
-
-getBlockSlotNo
-  :: CardanoBlock StandardCrypto
-  -> String
-getBlockSlotNo = \case
-  BlockByron blk ->
-    show $ unSlotNo $ blockSlot blk
-  BlockShelley (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockAllegra (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockMary (ShelleyBlock _ headerHash) ->
-    show headerHash
-  BlockAlonzo (ShelleyBlock _ headerHash) ->
-    show headerHash
+hashParser :: ReadM ChainPoint
+hashParser = eitherReader $
+  \chainPoint ->
+    case elemIndex ',' chainPoint of
+      Nothing  -> Left $ "Failed to parse chain point specification. The format "
+                      <> "should be HASH,SLOT."
+      Just idx ->
+        let (hash, slot') = splitAt idx chainPoint
+            slot :: Word64
+            slot = read (drop 1 slot') in
```

Comment:
with `readEither` we could propagate the error up to the ReadM

Reply:
I will use `either` to write this in a more monadic style.